OMCompiler/SimulationRuntime/c/simulation/solver/kinsol_b.c

```diff
@@ -1584,6 +1585,74 @@ static modelica_boolean nlsKinsolErrorHandler(int errorCode, DATA *data,
   return TRUE;
 }
 
+/**
+ * @brief Function dedicated for flag '--saveInitialGuess_system=/path/to/file.mat,nls_index' that computes only the
+ *        torn part of a given nonlinear system with index nls_index and writes it to a specified path. Will abort the simulation
+ *        if used incorrectly or if the file has been written.
+ *
+ * @param data                Runtime data struct.
+ * @param threadData          Thread data for error handling.
+ * @param nlsData             Pointer to non-linear system data.
+ * @return NLS_SOLVER_STATUS  Return NLS_SOLVED on success and NLS_FAILED otherwise.
+ */
+static void B_save_initial_guess_system(DATA *data, threadData_t *threadData, NONLINEAR_SYSTEM_DATA *nlsData, const char *arg)
+{
+  /* split the args of --saveInitialGuess_system (TODO: might want to seperate them into 2 flags) */
+  char buf[512];
+  strncpy(buf, arg, sizeof(buf));
+  buf[sizeof(buf)-1] = '\0';
+
+  char *comma = strchr(buf, ',');
+  if (!comma)
+  {
+    errorStreamPrint(OMC_LOG_STDOUT, 0, "Invalid format for --saveInitialGuess_system flag - Expected: '--saveInitialGuess_system=/path/to/file.mat,nls_index')");
+    exit(-1);
```

Comment:
I think we have `throwStreamPrint` that does more or less exactly this.

Reply:
Ah I see, if there is a try-catch it will probably not exit. Not sure how we (should) handle that.